internal/store/elasticsearch/discovery_search_repository_test.go

```diff
@@ -275,3 +275,107 @@ func loadTestFixture(cli *elasticsearch.Client, esClient *store.Client, filePath
 
 	return err
 }
+
+func TestSearcherGroup(t *testing.T) {
+	ctx := context.TODO()
+	t.Run("should return an error if group string array is empty", func(t *testing.T) {
+		cli, err := esTestServer.NewClient()
+		require.NoError(t, err)
+		esClient, err := store.NewClient(
+			log.NewNoop(),
+			store.Config{},
+			store.WithClient(cli),
+		)
+		require.NoError(t, err)
+
+		repo := store.NewDiscoveryRepository(esClient, log.NewNoop())
+		_, err = repo.GroupAssets(ctx, asset.GroupConfig{
+			GroupBy: []string{""},
+		})
+
+		assert.Error(t, err)
+	})
+
+	t.Run("fixtures", func(t *testing.T) {
+		cli, err := esTestServer.NewClient()
+		require.NoError(t, err)
+		esClient, err := store.NewClient(
+			log.NewNoop(),
+			store.Config{},
+			store.WithClient(cli),
+		)
+		require.NoError(t, err)
+
+		err = loadTestFixture(cli, esClient, "./testdata/search-test-fixture.json")
+		require.NoError(t, err)
+
+		repo := store.NewDiscoveryRepository(esClient, log.NewNoop())
+
+		type expectedRow struct {
+			GroupField []asset.GroupField
+			Asset      []asset.Asset
+		}
+		type groupTest struct {
+			Description string
+			Config      asset.GroupConfig
+			Expected    []expectedRow
```

Comment:
```suggestion
			Expected    []asset.GroupResult
```

not suer why we need to define another struct here.

Reply:
done